scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -54,8 +67,62 @@ void analyze() {
     @Override
     public void visit(CheckExpressionNode checkExpressionNode) {
         if (checkExpressionNode.checkKeyword().kind().equals(SyntaxKind.CHECKPANIC_KEYWORD)) {
-            scannerContext.getReporter().reportIssue(document, checkExpressionNode.location(),
-                    CoreRule.AVOID_CHECKPANIC.rule());
+            reportIssue(scannerContext, document, checkExpressionNode, CoreRule.AVOID_CHECKPANIC.rule());
+        }
+    }
+
+    @Override
+    public void visit(FunctionSignatureNode functionSignatureNode) {
+        functionSignatureNode.parameters().forEach(parameter -> {
```

Comment:
I feel like it would be cleaner if we break up the logic for extracting the parameters for each function "definition" (FunctinoSignatureNode, ImplicitAnonymousFunctionExpressionNode) and logic for testing seperatly. You can perhaps define two overloaded methods `private stream<Node> getParameters(FunctionSignatureNode|ImplicitAnonymousFunctionExpressionNode node)` for each then filter with isUnusedNode and call foreach reportIssue
```suggestion
        getParameters(functionSignatureNode).filter(node -> isUnusedNode(node, semnaticModel)).foreach(node -> reportIssue(scannerContext, document, node, CoreRule.UNUSED_FUNCTION_PARAMETERS.rule());
```

Reply:
Had a offline discussion with @heshanpadmasiri and decided to go with current appraoch.